### DISCUSSION TITLE
Linking with embedded modules in a static library.

### DISCUSSION THREAD

**Discussion by tdiff** [pybind/pybind11#4619] [Q&A]
Hi,

pybind registers module init functions in constructor of a global object (detail::embedded_module pybind11_module_<name>), defined with PYBIND11_EMBEDDED_MODULE.

Unfortunately, if the embedded module is defined in a static library that is linked to the main executable, the global object gets dropped by the linker (as it is not referenced elsewhere), so no initialization is performed and PyImport_AppendInittab is not called.

I must not be the first one to hit this (however, did not find smth similar in github issues), any advice would be much appreciated.

**Comment by allsey87** [comment]
I might have a (work in progress) solution to this. See https://github.com/pybind/pybind11/discussions/4796.

**Reply by ghost** [reply]
So is there a usable solution as of now ?
